src/ffi.rs

```diff
@@ -283,6 +311,93 @@ pub extern "C" fn line_sender_connect(
     host: *const libc::c_char,
     port: *const libc::c_char,
     err_out: *mut *mut line_sender_error) -> *mut line_sender
+{
+    line_sender_connect_secure(
+        net_interface,
+        host,
+        port,
+        std::ptr::null(),
+        err_out)
+}
+
+fn set_sec_opts(
+    builder: &mut super::LineSenderBuilder,
+    sec_opts: *const line_sender_sec_opts,
+    err_out: *mut *mut line_sender_error) -> bool
+{
+    if sec_opts.is_null() {
+        return true;
+    }
+
+    let auth_key_id = unsafe { (*sec_opts).auth_key_id };
+    let auth_priv_key = unsafe { (*sec_opts).auth_priv_key };
+    let auth_pub_key_x = unsafe { (*sec_opts).auth_pub_key_x };
+    let auth_pub_key_y = unsafe { (*sec_opts).auth_pub_key_y };
+
+    if auth_key_id.is_null() && auth_priv_key.is_null() &&
+       auth_pub_key_x.is_null() && auth_pub_key_y.is_null() {
+        return true;
```

Comment:
This is by design:
If a user wants to configure the authentication to be optional, they don't need to call different functions to create the connection, they can just call `line_sender_connect_secure`.
The previous `line_sender_connect` is there to avoid breaking the existing API.
That said, I have an API break coming up anyways (where the `..._name` type wil be split into two types).
Maybe this second function should just be simplified down to a single `line_sender_connect` function.

Reply:
> Maybe this second function should just be simplified down to a single `line_sender_connect` function.

Yeah, that would be nice.